spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/node/AgentLlmNode.java

```diff
@@ -387,13 +392,17 @@ private ChatClient.ChatClientRequestSpec buildChatClientRequestSpec(ModelRequest
 		List<Message> messages = appendSystemPromptIfNeeded(modelRequest);
 
 		List<ToolCallback> filteredToolCallbacks = filterToolCallbacks(modelRequest);
-		this.toolCallingChatOptions = ToolCallingChatOptions.builder()
-				.toolCallbacks(filteredToolCallbacks)
-				.internalToolExecutionEnabled(false)
-				.build();
+        if (modelRequest.getOptions() == null) {
+            this.chatOptions = ToolCallingChatOptions.builder()
+                .toolCallbacks(filteredToolCallbacks)
+                .internalToolExecutionEnabled(false)
+                .build();
+        } else {
+            this.chatOptions = modelRequest.getOptions();
+        }
```

Comment:
I think we need to copy and modify the original options from user to make sure it meets the requirements of the ReactAgent pattern. For example,

1. set internalToolExecutionEnabled to false
2. merge the toolcallbacks from ReactAgent.